edx_event_bus_kafka/internal/producer.py

```diff
@@ -177,18 +183,44 @@ def inner_to_dict(event_data, ctx=None):  # pylint: disable=unused-argument
     return key_serializer, value_serializer
 
 
+def get_headers_from_signal_and_metadata(signal: OpenEdxPublicSignal, event_metadata: EventsMetadata):
+    """
+    Create a dictionary of CloudEvent-compliant Kafka headers from an EventsMetadata object
+
+    Arguments:
+        event_metadata: An EventsMetadata object sent by an OpenEdxPublicSignal
+
+    Returns:
+        A dictionary of headers
+    """
+    if not event_metadata:
+        return {EVENT_TYPE_HEADER_KEY: signal.event_type.encode("utf-8")}
```

Comment:
1. I don't love the duplicated code.
2. I'm not clear on why the other non-metadata headers wouldn't always be included. I can see maybe leaving off specversion unless all CloudEvent required fields are there? But, do we want to return a partial header, or should we just be returning an error if it isn't provided?
```
header = {}

if event_metadata:
    header = {
        ...
    }

header[EVENT_TYPE_HEADER_KEY] = signal.event_type.encode("utf-8")
header[SPEC_VERSION_HEADER_KEY] = ...
header[CONTENT_TYPE_HEADER_KEY] = ...

return header
```

Reply:
I mostly did this for rollout purposes since I figured we'd probably want to make event_metadata mandatory. If we return an error when we don't have event_metadata we're setting ourselves up for a bunch of errors from studio if we don't upgrade very carefully.

Reply:
[wontfix unless you want to argue about it]

Reply:
Ok. If you see fit, you could copy the comment here as well:
```
# TODO: Make event_metadata required (https://github.com/openedx/openedx-events/issues/153)
```
Not a big deal either way, as long as you remember in the next ticket. Or, we could just add a link to this thread in that ticket. Or, you can just remember.